### PR TITLE
xtokens: ensuring exact error from xcm propagated

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
xcmp may fail because of dozens of reasons. it is very hard to debug and hard to setup. 
so solution is to be very specific about failure reason.

### CODE REVIEW

File: xtokens/src/lib.rs [hunk 1]
```diff
@@ -245,7 +330,7 @@ pub mod module {
 			let weight = T::Weigher::weight(&mut msg).map_err(|()| Error::<T>::UnweighableMessage)?;
 			T::XcmExecutor::execute_xcm_in_credit(origin_location, msg, weight, weight)
 				.ensure_complete()
-				.map_err(|_| Error::<T>::XcmExecutionFailed)?;
+				.map_err(xcm_to_pallet_error::<T>)?;
```
Comment (zjb0807) [inline xtokens/src/lib.rs:333]: I think it is enough to print the log here.

Comment (dzmitry-lahoda) [reply xtokens/src/lib.rs:333]: may be. 
may be you know how to get logs in xcm simulator tests https://stackoverflow.com/questions/70085232/how-to-see-logs-out-of-tests-running-substrate-cumulus-based-runtime ?

**Review by xlc — APPROVED** [review]
I am fine with this. Later can be updated once https://github.com/paritytech/substrate/pull/10242 is available

File: xtokens/src/lib.rs (hunk 1 above)
Comment (zjb0807) [reply xtokens/src/lib.rs:333]: 1. add `env_logger::init();` in the top of test method.
2. Run with
```
RUST_LOG=trace,parity-db=error,trie=error cargo test
```

**Comment by dzmitry-lahoda** [comment]
replaced with logger. added comment to use paritytech/substrate#10242  as soon as ready.

File: xtokens/src/lib.rs [hunk 2]
```diff
@@ -245,7 +247,10 @@ pub mod module {
 			let weight = T::Weigher::weight(&mut msg).map_err(|()| Error::<T>::UnweighableMessage)?;
 			T::XcmExecutor::execute_xcm_in_credit(origin_location, msg, weight, weight)
 				.ensure_complete()
-				.map_err(|_| Error::<T>::XcmExecutionFailed)?;
+				.map_err(|error| {
+					log::trace!("Failed execute transfer message with {:?}", error);
```
Comment (zjb0807) [inline xtokens/src/lib.rs:251]:
```suggestion
					log::error!(target: "xtokens", "Failed execute transfer message with {:?}", error);
```

**Pull request merged**